Android/Java/app/src/main/java/com/ironsource/ironsourcesdkdemo/DemoActivity.java

```diff
@@ -119,50 +144,68 @@ public void showRewardedVideoButtonTapped(View view) {
 
     public void loadInterstitialButtonTapped(View view) {
         // This will load an Interstitial ad
-
         log("loadInterstitial");
-        IronSource.loadInterstitial();
+        if (mInterstitialAd != null) {
+            mInterstitialAd.loadAd();
+        }
+    }
+
+    public void createInterstitialAd() {
+        mInterstitialAd = new LevelPlayInterstitialAd(INTERSTITIAL_AD_UNIT_ID);
+        mInterstitialAd.setListener(new DemoInterstitialAdListener(this));
+
+        setEnablementForButton(DemoButtonIdentifiers.LOAD_INTERSTITIAL_BUTTON_IDENTIFIER, true);
     }
 
     public void showInterstitialButtonTapped(View view) {
-        // It is advised to make sure there is available ad before attempting to show an ad
-        if (IronSource.isInterstitialReady()) {
+        // It is advised to make sure there is available ad that isn't capped before attempting to show it
+        if (mInterstitialAd != null && mInterstitialAd.isAdReady()) {
             // This will present the Interstitial.
             // Unlike Rewarded Videos there are no placements.
 
             log("showInterstitial");
-            IronSource.showInterstitial();
+            mInterstitialAd.showAd(this);
         } else {
             // load a new ad before calling show
         }
     }
 
     public void loadBannerButtonTapped(View view) {
-        // call IronSource.destroyBanner() before loading a new banner
-        if (bannerParentLayout != null) {
-            this.destroyBanner();
+        // Load a banner ad. If the "refresh" option is enabled in the LevelPlay dashboard settings, the banner will automatically refresh at the specified interval,
+        // otherwise, the banner will remain static until manually destroyed
+        log("loadBanner");
```

Comment:
In some cases, we log the message before the condition, and in others, we do not. I think we need to align that.